src/Nox.Generator/Application/Commands/UpsertEnumerationTranslationsCommand.template.cs

```diff
@@ -0,0 +1,109 @@
+{
+	public AppDbContext DbContext { get; }
+	private static readonly Nox.Types.CultureCode _defaultCultureCode = Nox.Types.CultureCode.From("{{codeGeneratorState.Solution.Application.Localization.DefaultCulture}}");
+
+	public {{upsertCommand}}HandlerBase(
+        AppDbContext dbContext,
+		NoxSolution noxSolution) : base(noxSolution)
+	{
+		DbContext = dbContext;
+	}
+
+	public virtual async Task<IEnumerable<{{enumAtt.EntityDtoNameForLocalizedEnumeration}}>> Handle({{upsertCommand}} command, CancellationToken cancellationToken)
+	{
+		cancellationToken.ThrowIfCancellationRequested();
+		await OnExecutingAsync(command);
+
+		cancellationToken.ThrowIfCancellationRequested();
+		await OnExecutingAsync(command);
+		
+		var cultureCodes = command.{{enumAtt.EntityDtoNameForLocalizedEnumeration}}s.DistinctBy(d=>d.CultureCode).Select(d=>CultureCode.From(d.CultureCode)).ToList();
+		
+		var localizedEntities = await DbContext.{{contextProperty}}Localized.Where(x => cultureCodes.Contains(x.CultureCode)).AsNoTracking().ToListAsync(cancellationToken);
+		
+		var entities = new List<{{enumEntity}}>();
+		
+		command.{{enumAtt.EntityDtoNameForLocalizedEnumeration}}s.Where(dto=> !localizedEntities.Any(e=>e.Id == Enumeration.FromDatabase(dto.Id) && e.CultureCode == CultureCode.From(dto.CultureCode))).ForEach(dto =>
+		{
+			var e = new {{enumEntity}} {Id = Enumeration.FromDatabase(dto.Id), CultureCode = CultureCode.From(dto.CultureCode), Name = dto.Name};
+			DbContext.Entry(e).State = EntityState.Added;
+			entities.Add(e);
+		});
+		
+		command.{{enumAtt.EntityDtoNameForLocalizedEnumeration}}s.Where(dto=> localizedEntities.Any(e=>e.Id == Enumeration.FromDatabase(dto.Id) && e.CultureCode == CultureCode.From(dto.CultureCode))).ForEach(dto =>
+		{
+			var e = new {{enumEntity}} {Id = Enumeration.FromDatabase(dto.Id), CultureCode = CultureCode.From(dto.CultureCode), Name = dto.Name};
+			DbContext.Entry(e).State = EntityState.Modified;
+			entities.Add(e);
+		});
+		
+		command.{{enumAtt.EntityDtoNameForLocalizedEnumeration}}s.Where(dto=>dto.CultureCode == _defaultCultureCode.Value).ForEach(dto =>
+		{
+			var e = new {{enumAtt.EntityNameForEnumeration}} { Id = Enumeration.FromDatabase(dto.Id), Name = dto.Name };
+			DbContext.Entry(e).State = EntityState.Modified;
+		});
+		
+
+		await OnBatchCompletedAsync(command, localizedEntities);
+		await DbContext.SaveChangesAsync(cancellationToken);
+		return command.{{enumAtt.EntityDtoNameForLocalizedEnumeration}}s;
+	}
+}
+public class {{upsertCommand}}Validator : AbstractValidator<{{upsertCommand}}>
+{
+	private static readonly Nox.Types.CultureCode _defaultCultureCode = Nox.Types.CultureCode.From("{{codeGeneratorState.Solution.Application.Localization.DefaultCulture}}");
+	private static readonly string[] _supportedCultureCodes = new string[] { {{- for cultureCode in codeGeneratorState.Solution.Application.Localization.SupportedCultures }} "{{cultureCode}}",{{- end}} };
```

Comment:
```suggestion
	private static readonly Nox.Type.CultureCode[] _supportedCultureCodes = new string[] { {{- for cultureCode in codeGeneratorState.Solution.Application.Localization.SupportedCultures }} Nox.Type.CultureCode.From("{{cultureCode}}"),{{- end}} };
```

Reply:
Since Dto contains string,  comparison  in rule uses string values as below;
RuleForEach(x => x.{{enumAtt.EntityDtoNameForLocalizedEnumeration}}s)
			.Must(x => _supportedCultureCodes.Contains(x.CultureCode))

Do we need to convert these strings into value objects and use theirs values again ?  What do you think ?

Reply:
Desired changes have been applied